walkthroughs/week-4-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -13,11 +13,54 @@
 // limitations under the License.
 
 package com.google.sps;
+import com.google.sps.FindMeetingQuery;
+import com.google.sps.MeetingRequest;
+import com.google.sps.TimeRange;
+import java.util.*;
 
-import java.util.Collection;
 
 public final class FindMeetingQuery {
+    Collection<TimeRange> timeRanges = new ArrayList<>();
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    if(request.getDuration() > TimeRange.WHOLE_DAY.duration()) {
+        return timeRanges;
+    }
+    if(request.getAttendees ().isEmpty() || request.getDuration() == 0 || events.isEmpty()) {
+        timeRanges.add(TimeRange.WHOLE_DAY);
+        return timeRanges;
+    }
+    HashSet<String> req_attendants = new HashSet<>();
+    for(String person: request.getAttendees())
+        req_attendants.add(person);
+    long req_duration = request.getDuration();
+
+    int start = 0;
+    int end = 0;
+    while(end <= TimeRange.END_OF_DAY){
+        if(available(events, req_attendants, end)) {
```

Comment:
alternative to consider: can this if statement be removed? It looks to me like the outer while loop will basically run the while(available) and while(!available) code blocks alternately. what if after running the while(available) block, the code just continued on to while(!available) instead of returning to the upper loop to re-check the status?

in general though, this is a nice solution! I like how you just have to run through the day once and not backtrack.